x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/index/IndexAuditTrail.java

```diff
@@ -277,7 +277,7 @@ private boolean canStart(ClusterState clusterState) {
         }
 
         if (TemplateUtils.checkTemplateExistsAndVersionMatches(INDEX_TEMPLATE_NAME, SECURITY_VERSION_STRING,
-                clusterState, logger, Version.CURRENT::onOrAfter) == false) {
+                clusterState, logger, Version.CURRENT::onOrBefore) == false) {
```

Comment:
bug 1.
new nodes, with a greater version, will not be stopped to create a rolled over index using an older template.

Reply:
Hmm, maybe I am misunderstanding but I don't follow the logic.

1. template does not exist so we should not start - this makes sense
2. template exists but is onOrBefore the current version of the node. On a new node, say 6.5.1, this means that if the version in the template is <= 6.5.1 that we start. Why would we start with a older template?

Reply:
I think the scenario you're describing is just the scenario for `onOrAfter`:
```
public boolean onOrAfter(Version version) {
  return version.id <= id;
}
```
In this case `id` is `6.5.1` and `version.id` is the version of the template, because the predicate is `Version.CURRENT::onOrAfter` .

As another anchor, further down in the function:
```
Version.fromString(versionString).onOrAfter(Version.CURRENT)
```
is correct, because it has the template version on the left of CURRENT.

You don't want to know for how long I had been banging my head on it....

Reply:
Gah! This is not easy to understand. For my sanity:

Version.CURRENT on new node = 6.5.1
template version = 6.5.0

Using onOrAfter would be:
`6.5.0 <= 6.5.1` = `true`

Using onOrBefore would be:
`6.5.0 >= 6.5.1` = `false`